src/client/src/tscSQLParser.c

```diff
@@ -5623,25 +5623,36 @@ int32_t doCheckForCreateFromStable(SSqlObj* pSql, SSqlInfo* pInfo) {
   SSchema* pTagSchema = tscGetTableTagSchema(pStableMeterMetaInfo->pTableMeta);
 
   STagData* pTag = &pCreateTable->usingInfo.tagdata;
-  char* tagVal = pTag->data;
+  SKVRowBuilder kvRowBuilder = {0};
+  if (tdInitKVRowBuilder(&kvRowBuilder) < 0) {
+    return TSDB_CODE_TSC_OUT_OF_MEMORY;
+  }
+
   int32_t ret = TSDB_CODE_SUCCESS;
-  
   for (int32_t i = 0; i < pList->nExpr; ++i) {
-    if (pTagSchema[i].type == TSDB_DATA_TYPE_BINARY || pTagSchema[i].type == TSDB_DATA_TYPE_NCHAR) {
+    SSchema* pSchema = pTagSchema + i;
+    if (pSchema->type == TSDB_DATA_TYPE_BINARY || pSchema->type == TSDB_DATA_TYPE_NCHAR) {
       // validate the length of binary
-      if (pList->a[i].pVar.nLen + VARSTR_HEADER_SIZE > pTagSchema[i].bytes) {
+      if (pList->a[i].pVar.nLen + VARSTR_HEADER_SIZE > pSchema->bytes) {
         return invalidSqlErrMsg(tscGetErrorMsgPayload(pCmd), msg3);
       }
     }
     
-    ret = tVariantDump(&(pList->a[i].pVar), tagVal, pTagSchema[i].type, true);
+    char tagVal[TSDB_MAX_TAGS_LEN];
+    ret = tVariantDump(&(pList->a[i].pVar), tagVal, pSchema->type, true);
     if (ret != TSDB_CODE_SUCCESS) {
       return invalidSqlErrMsg(tscGetErrorMsgPayload(pCmd), msg4);
```

Comment:
need to call tdDestroyKVRowBuilder(&kvRowBuilder); in error case for memory leakage